relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -192,24 +283,37 @@ impl SamplingRule {
 /// The fields will be used by rules to check if they apply.
 trait FieldValueProvider {
     /// gets the value of a field
-    fn get_value(&self, path: &str) -> FieldValue;
+    fn get_value(&self, path: &str) -> Value;
     /// what type of rule can be applied to this provider
     fn get_rule_type(&self) -> RuleType;
+    /// returns a filtering function for custom operators.
+    /// The function returned takes the provider and a condition definition and
+    /// returns a match result
+    fn get_custom_operator(
+        name: &str,
+    ) -> fn(condition: &CustomCondition, slf: &Self, ip_addr: Option<IpAddr>) -> bool;
```

Comment:
Probably also for a follow-up: Can we make it so that the Field provider inherently has access to the `ip_addr` (or, better, the entire RequestMeta)? We could implement the trait on a type like this:

```
struct EventValueProvider<'a> {
  event: &'a Annotated<Event>,
  meta: &'a RequestMeta,
}
```